### ISSUE TITLE
Build issue with master solution

### ISSUE THREAD

**Issue by tnguyen-Alta** [zeromq/libcurve#35]
I am new to zeromq. I was trying to follow the instruction however failed to build the libcurve with MSBuild (all version from VS2010, VS2012, VS2013 and VS2015). I have the following built success:
1. Libsodium
2. Libzmq - version 4.1
3. CZMQ - version 3.0

I would like to get your opinion on how to build libcurve in a proper way. I have attached the build log files for more details.

Thank you very much.

[build_10.txt](https://github.com/zeromq/libcurve/files/834639/build_10.txt)
[build_14.txt](https://github.com/zeromq/libcurve/files/834640/build_14.txt)

**Comment by bluca** [comment]
libzmq 4.1 has integrated curve support - why are you trying to use this library?

**Comment by tnguyen-Alta** [comment]
Hi @bluca , thanks for the prompt response.

Reason I tried libcurve because I thought in order to use zmq_setsockopt (*socket, ZMQ_CURVE_ROUTER, const *char, size_t), I would need to have ZMQ_HAVE_CURVE. The value I get from geterr() is 22 meaning that I didn't hit to :

"libzmq\src\option.cpp"

//  If curve encryption isn't built, these options provoke EINVAL
#ifdef ZMQ_HAVE_CURVE
        case ZMQ_CURVE_SERVER:
            if (is_int && (value == 0 || value == 1)) {
                as_server = value;
                mechanism = value? ZMQ_CURVE: ZMQ_NULL;
                return 0;
            }
            break;

**Comment by bluca** [comment]
That's because of a configuration error in your libzmq build. Make sure it can find libsodium and it should enable curve.

**Comment by tnguyen-Alta** [comment]
@bluca You are right. I missed to change libzmq Properties\Configuration Properties\ZMQ Option\Libsodium ON to Yes.

I am able to use CURVE now.

Thank you very much.

**Comment by bluca** [comment]
No problem

**Comment by tnguyen-Alta** [comment]
Hi Bluca,

I thought I have resolved this problem. I actually resolved the libsodium option with libzmq version 4.2 instead of 4.1. Now I am in need to build libzmq version 4.1 with ZMQ_HAVE_CURVE (libsodium) however I am still not able to have it enabled. Can you give me some instruction on this? In my Libzmq\Properties\Configuration Property, I have set ZMQ Options\sodium to Yes and apply to All Configurations. I also choose in Configuration Property\Local Dependencies\libsodium\Linkage=Dynamic (DLL).

Thank you for your supports.
